### DISCUSSION TITLE
Adding support for downstream quantification tools

### DISCUSSION THREAD

**Discussion by wfondrie** [wfondrie/mokapot#22] [Ideas]
On Twitter, Sam Payne asked whether mokapot supports downstream quanitation tools, such as [FlashLFQ](https://github.com/smith-chem-wisc/FlashLFQ) and [IonQuant](https://github.com/Nesvilab/IonQuant). Here's the [tweet](https://twitter.com/byu_sam/status/1369675514693095425?s=20).

The current release version of mokapot does not easily support these tools, but it is certainly needed. The difficulty arises from the need for additional metadata---beyond what is typically provided to Percolator---to connect PSMs with their corresponding spectra in the mass spectrometry data files. 

I'm working on solving this in the `quant` branch, where I've added optional parameters to the `read_pin()` function for the necessary columns and automatically extracted this data when the `read_pepxml()` is used. Although still experimental, these features can be tried by installing the `quant` branch directly from GitHub:

```
pip install git+https://github.com/wfondrie/mokapot@quant
```

Once installed, an input file for FlashLFQ can be generated from a PepXML file with the following:
```Python
import mokapot

psms = mokapot.read_pepxml("psms.pep.xml")
results, models = mokapot.brew(psms)
results.to_flashlfq()
```

This feature needs to be formally tested, but it has worked well with the ad-hoc tests I've performed.

**Comment by wfondrie** [comment]
This, along with a few other new features, is now available in mokapot v0.7.1 🎉

**Comment by jspaezp** [comment]
Hey there @wfondrie .... its me again ... lol.
Is there any design choice behind having the flaslfq output only available though the python interface or could we talk about adding a flag to the command line interface to save that was well or save as that format?
Thanks a lot again!

**Reply by wfondrie** [reply]
I'm glad you're finding mokapot useful! This is exactly the kind of thing this discussion board is for. 😄 

The main reason that the FlashLFQ output is only available through the Python API is because it requires additional information that is not normally available in the PIN files that are created for Percolator. To generate the FlashLFQ output, we need to know the MS data file and the specific scan from which a PSM was found. This information is readily available to search engines, but alas, is often not included for the post processing step. 

 My reason against doing so was mainly that we'd need to add a significant number of new command line arguments to support it, which may make the CLI pretty confusing to use.
